### PR TITLE
Deprecated methods that have better replacements on Map Read API

### PR DESCRIPTION
**Pull request by mhutchinson**
This is a fairly hard deprecation where we will migrate all known clients over and delete these methods. Having SetLeaves in its current form will block us from moving to async tree synthesis after WriteLeaves.

- [x] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly (including the [feature implementation matrix](docs/Feature_Implementation_Matrix.md)).

### CODE REVIEW

File: trillian_map_api.proto [hunk 1]
```diff
@@ -193,20 +193,22 @@ service TrillianMap {
   rpc GetLeafByRevision(GetMapLeafByRevisionRequest) returns (GetMapLeafResponse) {}
   rpc GetLeaves(GetMapLeavesRequest) returns (GetMapLeavesResponse) {}
   rpc GetLeavesByRevision(GetMapLeavesByRevisionRequest) returns (GetMapLeavesResponse) {}
-  // GetLeavesByRevisionNoProof returns the requested map leaves without inclusion proofs.
-  // This API is designed for internal use where verification is not needed.
-  rpc GetLeavesByRevisionNoProof(GetMapLeavesByRevisionRequest) returns (MapLeaves) {}
+  // Deprecated: this should only be used be writers, which should migrate
```
Comment (pavelkalinnikov) [inline trillian_map_api.proto]: typo: used by?

passim

**Comment by pavelkalinnikov** [comment]
Tests are failing because the deprecated method is still used in hammer and some tests.

**Comment by mhutchinson** [comment]
This is blocked on #1872 and 1875 to remove the usages of the deprecated methods. If we can't get those merged soon, then the other option is to disable the lint check for these usages.

**Comment by mhutchinson** [comment]
I had to disable one of the lint checks for a single line - I don't want to change the API surface area here, so disabling this check is the only pragmatic way to do that while still maintaining functionality.

**Pull request merged**